pkg/exporter/sumologicexporter/config.go

```diff
@@ -83,18 +83,18 @@ func CreateDefaultHTTPClientSettings() confighttp.HTTPClientSettings {
 func (cfg *Config) Validate() error {
 
 	switch cfg.CompressEncoding {
-	case configcompression.Gzip:
-	case configcompression.Deflate:
+	case configcompression.TypeGzip:
```

Comment:
Changes that account for upstream deprecations

Reply:
Can you submit these in a separate PR? Would make this one easier to review.